static/js/containers/PostPage.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+import React from 'react';
+import { connect } from 'react-redux';
+import R from 'ramda';
+
+import Card from '../components/Card';
+import Loading, { anyProcessing, allLoaded } from '../components/Loading';
+import ChannelBreadcrumbs from '../components/ChannelBreadcrumbs';
+import PostDisplay from '../components/PostDisplay';
+
+import { actions } from '../actions';
+
+import type { Dispatch } from 'redux';
+import type { Match } from 'react-router';
+
+class PostPage extends React.Component {
+  props: {
+    match:            Match,
+    dispatch:         Dispatch,
+    posts:            Object,
+    channels:         Object,
+  };
+
+  getMatchParams = () => {
+    const { match: { params }} = this.props;
+    return [params.postID, params.channelName];
+  }
+
+  updateRequirements = () => {
+    const { dispatch, channels, posts } = this.props;
+    const [postID, channelName] = this.getMatchParams();
+
+    if (R.isNil(posts.data.get(postID))) {
```

Comment:
We will only get to this function if a request for the `Post` API has already been made. The trick thing is that the request in question could have been for a *different* post than the one we're currently interested in. Consider this case:

- navigate to `/channel/foobar/postid`
- navigate back to `/channel/foobar`
- navigate to `/channel/foobar/anotherpostid`

at this point we have already hit the posts API once (to fetch `postid`). when we load the `PostPage` component, to try to view `anotherpostid`, we will therefore already have `loaded === true`, but we can't quite trust it. It seems, in my testing, that there is typically 1 render call on the changed component before the new request (to get `anotherpostid`) kicks off, when we will have the case that `processing` is false, `loaded` is true, and the data for `anotherpostid` hasn't yet been fetch. So I added this check just to make sure the data is actually there before we try to render anything with it.

Does that make sense? I ran into the same issue with navigating around between subreddits when I was working on the frontpage UI, so I added an equivalent check to the `ChannelPage` container then.